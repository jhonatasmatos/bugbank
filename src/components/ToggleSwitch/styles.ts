import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
