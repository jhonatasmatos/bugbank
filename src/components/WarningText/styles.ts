import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
