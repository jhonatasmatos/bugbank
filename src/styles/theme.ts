export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
