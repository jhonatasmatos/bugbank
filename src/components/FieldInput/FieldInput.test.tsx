// import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
// import { theme } from '../../styles/theme';
import { FieldInput } from '.';

const fn = jest.fn();

describe('<FieldInput />', () => {
  it('should render with success', () => {
    const { container } = renderTheme(
      <FieldInput
        label="test"
        name="test"
        visible={true}
        type="text"
        placeholder="test"
        messageError="error"
        register={fn}
      />,
    );

    expect(container).toBeInTheDocument();
  });

  it('should render with password type and not visible', () => {
    const { container } = renderTheme(
      <FieldInput
        label="test"
        name="test"
        visible={false}
        type="password"
        placeholder="test"
        messageError="error"
        register={fn}
      />,
    );

    const input = container.querySelector('input');
    const warningText = container.querySelector('p');

    expect(input).toHaveProperty('type', 'password');
    expect(warningText).toHaveStyle('opacity: 0');
  });

  it('should render with success', () => {
    const { container } = renderTheme(
      <FieldInput
        label="test"
        name="test"
        visible={true}
        type="text"
        placeholder="test"
        messageError="error"
        register={fn}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
