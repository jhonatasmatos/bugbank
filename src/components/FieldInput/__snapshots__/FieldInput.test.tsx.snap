// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldInput /> should render with success 1`] = `
.c0 {
  position: relative;
  margin-bottom: 8px;
  padding-bottom: 16px;
}

.c0 .input__label {
  font-size: 1.8rem;
  font-family: 'Nunito',sans-serif;
  color: #D62A92;
}

.c0 .input__default {
  width: 100%;
  padding: 1.3rem 1rem;
  border: 1px solid #D62A92;
  border-radius: 0.4rem;
  font-size: 1.6rem;
  margin-top: 0.6rem;
  box-shadow: 0px 0px 0px 0px #D62A92;
}

.c0 .input__default:focus {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  box-shadow: 0px 0px 0px 1px #D62A92;
}

.c0 .input__warging {
  opacity: 1;
  bottom: -25px;
  position: absolute;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  color: red;
  font-size: 1.2rem;
  font-family: 'Nunito',sans-serif;
}

<div>
  <div
    class="c0 input__child"
  >
    <label
      class="input__label"
      for="test"
    >
      test
    </label>
    <input
      autocomplete="off"
      class="input__default"
      placeholder="test"
      type="text"
    />
    <p
      class="input__warging"
    >
      error
    </p>
  </div>
</div>
`;
